arangod/RocksDBEngine/RocksDBMetaCollection.cpp

```diff
@@ -1661,63 +1661,38 @@ Result RocksDBMetaCollection::applyUpdatesForTransaction(
 
 /// @brief lock a collection, with a timeout
 ErrorCode RocksDBMetaCollection::doLock(double timeout, AccessMode::Type mode) {
-  uint64_t waitTime = 0;  // indicates that time is uninitialized
   double startTime = 0.0;
 
   // user read operations don't require any lock in RocksDB, so we won't get
   // here. user write operations will acquire the R/W lock in read mode, and
   // user exclusive operations will acquire the R/W lock in write mode.
   TRI_ASSERT(mode == AccessMode::Type::READ || mode == AccessMode::Type::WRITE);
 
-  while (true) {
-    bool gotLock = false;
-    if (mode == AccessMode::Type::WRITE) {
-      gotLock = _exclusiveLock.tryLockWrite();
-    } else if (mode == AccessMode::Type::READ) {
-      gotLock = _exclusiveLock.tryLockRead();
-    } else {
-      // we should never get here
-      TRI_ASSERT(false);
-      return TRI_ERROR_INTERNAL;
-    }
-
-    if (gotLock) {
-      // keep the lock and exit the loop
-      return TRI_ERROR_NO_ERROR;
-    }
-
-    double now = TRI_microtime();
-    if (waitTime == 0) {  // initialize times
-      // set end time for lock waiting
-      if (timeout <= 0.0) {
-        timeout = defaultLockTimeout;
-      }
+  bool gotLock = false;
+  if (mode == AccessMode::Type::WRITE) {
+    gotLock = _exclusiveLock.try_lock_for(std::chrono::duration<double>(timeout));
+  } else if (mode == AccessMode::Type::READ) {
+    gotLock = _exclusiveLock.try_lock_shared_for(std::chrono::duration<double>(timeout));
+  } else {
+    // we should never get here
+    TRI_ASSERT(false);
+    return TRI_ERROR_INTERNAL;
+  }
 
-      startTime = now;
-      waitTime = 1;
-    } else {
-      TRI_ASSERT(startTime > 0.0);
+  if (gotLock) {
+    // keep the lock and exit the loop
```

Comment:
We don't have a loop anymore
```suggestion
    // keep the lock and exit
```